homeassistant/helpers/entity_component.py

```diff
@@ -131,13 +108,68 @@ def _setup_platform(self, platform_type, platform_config,
         if platform is None:
             return
 
+        if self.group is None and self.group_name is not None:
```

Comment:
The reason this logic used to be in `add_entities` was that it would only create a group if a platform would actually add an entity (which can happen that they don't).